read_buffer/src/lib.rs

```diff
@@ -470,20 +474,42 @@ impl Database {
     pub fn column_names(
         &self,
         partition_key: &str,
+        table_name: &str,
         chunk_ids: &[u32],
         predicate: Predicate,
     ) -> Result<RecordBatch> {
-        // Find all matching chunks using:
-        //   - time range
-        //   - measurement name.
-        //
-        // Execute query against matching chunks. The `tag_keys` method for
-        // a chunk allows the caller to provide already found tag keys
-        // (column names). This allows the execution to skip entire chunks,
-        // tables or segments if there are no new columns to be found there...
-        Err(Error::UnsupportedOperation {
-            msg: "`column_names` call not yet hooked up".to_owned(),
-        })
+        let partition_data = self.data.read().unwrap();
+
+        let partition = partition_data
+            .partitions
+            .get(partition_key)
+            .ok_or_else(|| Error::PartitionNotFound {
+                key: partition_key.to_owned(),
+            })?;
+
+        let chunk_data = partition.data.read().unwrap();
+        let mut filtered_chunks = vec![];
```

Comment:
I see you need to use a `for` loop rather than `map`/`collect` so that you can use `?`, but I think this could save some potential reallocations by at least doing:

```suggestion
        let mut filtered_chunks = Vec::with_capacity(chunk_ids.len());
```